### PR TITLE
bugfix/v1-099

### PR DESCRIPTION
**Pull request by Julia-Maliutsina**
1. Search results were displayed after 1st render of page - fixed.
2. Added formatting for text that can be pasted it search input.
3. Disallowed entering whitespace as a first symbol.

### CODE REVIEW

File: frontend/src/components/Layout/Header/SearchCourses/SearchCoursesContainer.tsx [hunk 1]
```diff
@@ -15,12 +15,14 @@ const SearchCoursesContainer: React.FC = () => {
   useEffect(() => {
     const timer = setTimeout(() => {
       const getCourses = async (search: string) => {
-        const searchResponse = await searchAllCourses(search);
-        if (searchResponse.data) {
-          setCoursesFound(searchResponse.data);
-          setSearchOpen(true);
-        } else {
-          enqueueSnackbar('Something went wrong');
+        if (searchInputValue.length > 0) {
```
Comment (irina-sinitsyna) [inline frontend/src/components/Layout/Header/SearchCourses/SearchCoursesContainer.tsx:18]: Remove magic number please

File: frontend/src/components/Layout/Header/SearchCourses/SearchCoursesContainer.tsx [hunk 2]
```diff
@@ -15,12 +15,14 @@ const SearchCoursesContainer: React.FC = () => {
...
+          const searchResponse = await searchAllCourses(search);
+          if (searchResponse.data) {
+            setCoursesFound(searchResponse.data);
+            setSearchOpen(true);
+          } else {
+            enqueueSnackbar('Something went wrong');
```
Comment (irina-sinitsyna) [inline frontend/src/components/Layout/Header/SearchCourses/SearchCoursesContainer.tsx:24]: Here you should also add snackbar type(successSnackbar or errorSnackbar) from constants/snackbarVariant and also move the message to this file too
So you will have something like that: `enqueueSnackbar(errorSnackbarMessage.failed, errorSnackbar);`

File: frontend/src/components/Layout/Header/SearchCourses/SearchCoursesContainer.tsx [hunk 3]
```diff
@@ -30,7 +32,8 @@ const SearchCoursesContainer: React.FC = () => {
 
   const searchCourses = (event: React.ChangeEvent<HTMLInputElement>): void => {
     if (event.target.value.length) {
-      setSearchInputValue(event.target.value);
+      const formattedValue = event.target.value.split(/\s+/).join(' ').trimStart().trimEnd();
```
Comment (irina-sinitsyna) [inline frontend/src/components/Layout/Header/SearchCourses/SearchCoursesContainer.tsx:35]: I think we should move it to helpers

File: frontend/src/components/Layout/Header/SearchCourses/SearchCoursesContainer.tsx [hunk 4]
```diff
@@ -40,12 +43,21 @@ const SearchCoursesContainer: React.FC = () => {
     setSearchOpen(false);
   };
 
+  const checkSpace = (event: React.KeyboardEvent) => {
+    const { key } = event;
+    const formattedKey = key.trim();
+    if (!formattedKey && searchInputValue.length === 0) {
```
Comment (irina-sinitsyna) [inline frontend/src/components/Layout/Header/SearchCourses/SearchCoursesContainer.tsx:49]: Magic number and looks like a helper again

File: frontend/src/components/Layout/MobileSearch/MobileSearchContainer.tsx [hunk 5]
```diff
@@ -16,12 +16,14 @@ const MobileSearchContainer: React.FC = () => {
...
-          setSearchOpen(true);
-        } else {
-          enqueueSnackbar('Something went wrong');
+        if (searchInputValue.length > 0) {
+          const searchResponse = await searchAllCourses(search);
+          if (searchResponse.data) {
+            setCoursesFound(searchResponse.data);
+            setSearchOpen(true);
+          } else {
+            enqueueSnackbar('Something went wrong');
```
Comment (irina-sinitsyna) [inline frontend/src/components/Layout/MobileSearch/MobileSearchContainer.tsx:25]: Please change it according to previous comment

File: frontend/src/components/Layout/MobileSearch/MobileSearchContainer.tsx [hunk 6]
```diff
@@ -16,12 +16,14 @@ const MobileSearchContainer: React.FC = () => {
   useEffect(() => {
     const timer = setTimeout(() => {
       const getCourses = async (search: string) => {
-        const searchResponse = await searchAllCourses(search);
-        if (searchResponse.data) {
-          setCoursesFound(searchResponse.data);
-          setSearchOpen(true);
-        } else {
-          enqueueSnackbar('Something went wrong');
+        if (searchInputValue.length > 0) {
```
Comment (irina-sinitsyna) [inline frontend/src/components/Layout/MobileSearch/MobileSearchContainer.tsx:19]: Magic number

File: frontend/src/components/Layout/MobileSearch/MobileSearchContainer.tsx [hunk 7]
```diff
@@ -31,7 +33,8 @@ const MobileSearchContainer: React.FC = () => {
 
   const searchCourses = (event: React.ChangeEvent<HTMLInputElement>): void => {
     if (event.target.value.length) {
-      setSearchInputValue(event.target.value);
+      const formattedValue = event.target.value.split(/\s+/).join(' ').trimStart().trimEnd();
```
Comment (irina-sinitsyna) [inline frontend/src/components/Layout/MobileSearch/MobileSearchContainer.tsx:36]: Helpers?

File: frontend/src/components/Layout/MobileSearch/MobileSearchContainer.tsx [hunk 8]
```diff
@@ -40,12 +43,21 @@ const MobileSearchContainer: React.FC = () => {
   const handleSearchClose = () => {
     setSearchOpen(false);
   };
+
+  const checkSpace = (event: React.KeyboardEvent) => {
+    const { key } = event;
+    const formattedKey = key.trim();
```
Comment (irina-sinitsyna) [inline frontend/src/components/Layout/MobileSearch/MobileSearchContainer.tsx:49]: Magic number and looks like a helper again

**Review by irina-sinitsyna — CHANGES_REQUESTED** [review]

File: frontend/src/components/Layout/MobileSearch/MobileSearchContainer.tsx (hunk 6 above)
Comment (k-krasilnikova) [reply frontend/src/components/Layout/MobileSearch/MobileSearchContainer.tsx:19]:
```suggestion
        if (searchInputValue.length) {
```

**Review by irina-sinitsyna — APPROVED** [review]

**Review by k-krasilnikova — APPROVED** [review]

**Pull request merged**